src/Elastic.Apm/Filters/TransactionIgnoreUrlsFilter.cs

```diff
@@ -15,8 +16,20 @@ namespace Elastic.Apm.Filters
 	internal class TransactionIgnoreUrlsFilter
 	{
 		private readonly IConfigSnapshot _configSnapshot;
+
 		public TransactionIgnoreUrlsFilter(IConfigSnapshot configSnapshot) => _configSnapshot = configSnapshot;
-		public ITransaction Filter(ITransaction transaction) =>
-			WildcardMatcher.IsAnyMatch(_configSnapshot.TransactionIgnoreUrls, transaction?.Context?.Request?.Url?.PathName) ? null : transaction;
+
+		public ITransaction Filter(ITransaction transaction)
+		{
+			if (transaction is Transaction realTransaction)
+			{
+				// If there is no context, there is no URL either, therefore this transaction can't be filtered based on the URL.
+				if (!realTransaction.IsContextCreated)
```

Comment:
can combine nested `if`s?

Reply:
In https://github.com/elastic/apm-agent-dotnet/pull/1087 I added more to the outer if block: https://github.com/elastic/apm-agent-dotnet/pull/1087/files#diff-78983cac62ab3ac9a4a1ad0e2e32e6f04cb7b744979b4c8f9e3106b0fa5ef262R25 

Otherwise I'd agree :)